### PR TITLE
interfaces: add history-related (what history?) interface

### PR DESCRIPTION
**Pull request by renatofilho**
Create history interface.

History interface is used by telephony apps to retrieve the call log and messaging history.

### CODE REVIEW

File: interfaces/builtin/history.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+func (iface *HistoryInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	if iface.Name() != slot.Interface {
+		panic(fmt.Sprintf("slot is not of interface \"%s\"", iface.Name()))
+	}
+	return nil
+}
+
+func (iface *HistoryInterface) LegacyAutoConnect() bool {
```
Comment (niemeyer) [inline interfaces/builtin/history.go]: I believe those are now obsolete and can be dropped altogether.

File: interfaces/builtin/history.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+</policy>
+<policy context="default">
+    <deny send_destination="com.canonical.HistoryService"/>
+</policy>
+`)
+
+type HistoryInterface struct{}
+
+func (iface *HistoryInterface) Name() string {
+	return "history"
```
Comment (niemeyer) [inline interfaces/builtin/history.go]: We need something more clear and specific as a name for this one. Imagine a snap in the wild including `plugs: [history]`. What is the history service?  History of what?

File: interfaces/builtin/history.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	path=/org/freedesktop/Telepathy/Client,
+dbus (receive, send)
+	bus=session
+	path=/org/freedesktop/Telepathy/Client/HistoryDaemonObserver,
+dbus (receive, send)
+	bus=session
+	path=/org/freedesktop/Telepathy/AccountManager,
+`)
+
+var historyConnectedPlugAppArmor = []byte(`
```
Comment (niemeyer) [inline interfaces/builtin/history.go]: Looks reasonable on a quick look, but please get a full review from @jdstrand before merging.

**Comment by chipaca** [comment]
closing this, as it has conflicts and hasn't been kept up to date for months, nor has @renatofilho indicated that they have resolved the issues raised by @niemeyer. Feel free to reopen once these things are addressed.

**Pull request closed without merging**